### ISSUE TITLE
[Feature Request] Copy item info to clipboard via ^c similar to in-game

### ISSUE THREAD

**Issue by trackpete** [Stickymaddness/Procurement#151]
I'm working on a new version of my price macro that can do rudimentary matching for rares and other items and works by taking a [full item copy/paste submission for the parsing](http://exiletools.com/price-lookup-test.html) (so it's all done server side). 

I was thinking that if you could add similar text output functionality to Procurement on mouse-over and ^c, I could modify the macro to also work within Procurement. It doesn't have to be a perfect clone of in-game (obviously it won't have Item Level/etc.), but just the basic layout as already shown by the javascript renderer would be fantastic. 

Right now when I'm setting prices I have to have an in-game window open to run the macro on the items, but it'd be great to be able to do it straight from Procurement. Bonus: This wouldn't require you to actually interface with my price macro/upcoming API.

Cheers,

 -Pete

**Comment by Stickymaddness** [comment]
Hi Pete,

I'm still very keen to add this functionality to Procurement, the previous pull request kind of fell by the wayside. What is the current state of your API? 

Do you have an endpoint I can call with some rudimentary item data that will bring back pricing info? As you mentioned, the only limitation should be item level.

**Comment by trackpete** [comment]
I am currently in development on a whole new system, however you can take a look at an early version (I am really hoping to push this out in the next couple weeks), linked above. It's really simple, to hit the dev version basically send a POST request to http://api.exiletools.com/cgi-bin/dev-item-report-text with the following:

v=5
league=[league - warbands|tempest|hardcore|softcore]
itemData=[url encoded text data for an item as seen in-game]

There's also a fully functional version available at api.exiletools.com/item-report-text that works the same way, but it currently only operates on Gems and Uniques.

Right now the data returned is purely text output, intended for display in a popup. I will be adding a JSON output format in the near future, but right now I'm working hard to get my indexer handling all those stupid Acquisition threads that aren't getting bumped. ;(

tl;dr: old price API is completely gone, but the new one is way better, just need to finish work on it, which will hopefully be very soon.

**Comment by Stickymaddness** [comment]
I tried pasting text for rares in http://exiletools.com/price-lookup-test.html and making a post request to http://api.exiletools.com/cgi-bin/dev-item-report-text, but I get the following for any items I try:

Sorry! 0 out of 1420533 verified items with prices
updated in the last 7 days matched this search.

Tested with rare Sharkskin Gloves and Two-Stone ring.

**Comment by trackpete** [comment]
Yeah, that's one of the problems with the current logic, since it's a similarity comparison - it basically does a comparison of the current stats +/- 20%. If you don't get any results back, there's nothing like it for sale, which can mean it's a poor item or (more likely) it has one unusual stat, probably a bit low, unlike any other item.

It does have a *little* logic in that it searches for psuedo resists and totals, but it's very easy to throw it off. The most common examples is a ring with great stats and then like +3 to Dexterity, which won't be similar to ANY other item.

It's kinda ridiculous how much RNG is on POE, that you can search for an item and literally out of 1.4 million items in the DB, there isn't a single one with similar stats +/- 20%, but this happens a lot for anything other than common good items.

Try it again with something with more common/typical stats, or try manually deleting a line or two from the paste before you shove it into the form and see what happens!
